multinode/bootstrapping.md

```diff
@@ -0,0 +1,201 @@
+the data node, so the next step is ensuring that there is a valid
+extension installed in the database by either:
+
+* creating the extension, or
+
+* verifying that the extension is compatible with the extension
+  installed on the access node.
+
+### Creating the schema for the extension
+
+To install the extension, a schema is created to store the objects
+that the `timescaledb` extension needs. The schema name is decided by
+looking at what schema name is used for the extension on the access node and
+using the same name on the data node. If you have created the
+`timescaledb` extension in the default manner, `public` will be used
+for the schema on the access node, hence `public` will be used on the
+data node as well.
+
+If the schema name for the extension on the access node is `public`,
+no new schema is created, but otherwise it is expected that the schema
+name does *not* exist. In the event that the schema name exists (but
+is not `public`), an error will be raised and the bootstrapping will abort.
+
+Alternative description:
+> If the schema already exists on the data node the access node will
+> verify that it is correctly configured and proceed with installing
+> the extension.
+
+To change the schema used on the access node and data node, you can
+create a new schema, then create the extension with the schema that
+you want:
+
+```
+CREATE SCHEMA magic;
+CREATE EXTENSION timescaledb SCHEMA magic;
+```
+
+When validating the schema, it is necessary that the schema is owned
+by the current user, or is `public`.
+
+>:TIP: This will place all the Timescale functions (for example,
+> `create_hypertable` in the `magic` schema), so you need to qualify
+> all function references with `magic`. For example:
+>
+> ```
+> SELECT * FROM magic.create_hypertable('conditions', 'time', 3);
+> ```
+
+### Installing the extension
+
+After the schema is created, the extension is installed using the
+`CREATE EXTENSION` command with the `SCHEMA` option to refer to the
+schema from the section above. The schema name for the extension will
+be either `public` or a dedicated schema name.
+
+If the extension already exists in the database, an error will be
+raised and the bootstrapping aborted.
+
+Proposed changes:
+> If the extension already exists, it is validated to ensure that it
+> can be used with the access node and the procedure proceeds to set
+> the metadata.
+
+### Validating the extension
+
+Validation of the extension checks that the extension on the data node
+is of a version that is compatible with the version on the access node
+and that the access node has permissions to use the extension.
+
+The extension on the data node is compatible with the extension on the
+access node if the major versions are the same and the minor version
+on the data node is before or the same as the minor version on the access
+node. For example:
+
+- Compatible: data node v1.3.1, access node v1.5.1
```

Comment:
The policy for extension compatibility seems a lot more important than the process of bootstrapping in general. Can we try to emphasize the policy and reduce the process stuff?  Another question is if this validation is specific to bootstrapping? Shouldn't it happen on every connection or at least also during upgrades (i.e., when the data node already exists)? I am asking because the information here seems important outside the context of bootstrapping (e.g., as part of a section on node compatibility or upgrading). Now this important information is buried in a lot of bootstrapping implementation detail.

Generally, I think it is best to focus on documenting _what_ we do and _why_, instead of _how_ we do it.

Reply:
Yes, the verification happens each time a connection is established, so it is not restricted to only the bootstrapping.

I think this is a description of what we do and why? Not sure what you think should be written differently.

Reply:
Just saying that extension compatibility seems more important than other minute details here which warrants lifting it up from being buried in these bootstrapping details. And, further, node validation doesn't seem to be specific to bootstrapping. So, my comment is about structure rather than content.

In other words, these two things (extension version compatibility and validation) are connected. I could imagine, e.g., a separate (more front-and-center) doc/chapter on requirements/prerequisites/policy related to node interactions that are relevant to both upgrades and bootstrapping that doesn't require digging into the details of the bootstrapping process. If I imagine myself wearing the hat of a cluster administrator I would personally prefer a concise chapter specifically on node requirements and compatibility. This is crucial for me to conceptualize how to organize and maintain a cluster, without having to care about the details of the bootstrapping process.

So, what I am proposing is organizing this differently to aid readers by separating key information from minute details. Concretely we'd separate requirements/prerequisites/policy from bootstrapping details. For example, if you write "Validation of the extension checks that the extension on the data node
is of a version that is compatible with the version on the access node" you need not list the exact policy for compatibility, including examples here. Just reference another chapter that clearly and concisely deals with that, and which we can easily index in the TOC and reference from other places in the doc (e.g., upgrade docs).

But maybe this is part of a larger docs conversation that is better suited in another forum? I just worry that this will never happen if we just let the docs grow organically without thinking about the larger organization and structure.

Reply:
I think it is good to make the information available to the user, but I agree that we do need to look over the high-level structure. I think we can move the information as a separate PR for creating a description about validation.